### DISCUSSION TITLE
Filtering on XML attributes in v2

### DISCUSSION THREAD

**Discussion by dkx0stv** [TomWright/dasel#436] [Q&A] [answered]
Good afternoon, everybody!

This is probably trivial, but I am struggling with filtering on XML attributes in v2 & couldn't find any information - neither in the v2 documentation nor here.

A simple syntax example, like the one in the [v1 documentation](https://daseldocs.tomwright.me/v1/examples/xml#filtering-on-attributes), would help me out a ton.
```dasel -p xml '.data.users.(-primary=true).name'``` works like a charm in v1, whereas the result in v2 is an error message:
```Error: exit status 1: Error: could not resolve selector: bad syntax: function name required before open bracket, around (-primary=true)```

Thank you
Martin

**Comment by dkx0stv** [comment, marked as answer]
```dasel -r xml '.data.users.all().filter(equal(-primary,true)).name'```

Sorry if I wasted anybody's time.

**Reply by TomWright** [reply]
Thanks for posting the solution you found.

For anyone else in the same boat, XML attributes are just prepended with a `-` and are otherwise exactly the same as normal fields.
